kolibri/plugins/management/assets/src/state/facilityConfigPageActions.js

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable prefer-arrow-callback */
+const omit = require('lodash/fp/omit');
+const {
+  FacilityResource,
+  FacilityDatasetResource,
+} = require('kolibri').resources;
+const ConditionalPromise = require('kolibri.lib.conditionalPromise');
+const { samePageCheckGenerator } = require('kolibri.coreVue.vuex.actions');
+const preparePage = require('./preparePage');
+const { PageNames, defaultFacilityConfig, notificationTypes } = require('../constants');
+
+// When app is installed, the Facility is assigned id of `1`. Hardcoded here for now.
+const FACILITY_ID = 1;
+
+function resolveOnlyIfOnSamePage(promises, store) {
+  const ident = x => x;
+  return ConditionalPromise.all(promises)
+  .only(samePageCheckGenerator(store), ident, ident)._promise;
+}
```

Comment:
I've had to use the `_promise` escape hatch in a number of places because the chaining semantics of ConditionalPromise don't seem to match the Promise spec, esp. when it comes to error handling via `catch`.

I added some explanation in a comment, and maybe could look into extending ConditionalPromise later, although I'm more in favor of using something more standard like cancellable promises.